charms/worker/k8s/src/charm.py

```diff
@@ -212,6 +217,16 @@ def _apply_proxy_environment(self):
             with open("/etc/environment", mode="w", encoding="utf-8") as file:
                 file.write("\n".join([f"{k}={v}" for k, v in current_env.items()]))
 
+    def _generate_unique_cluster_name(self) -> str:
+        """Use a stable input to generate a unique cluster name.
+
+        Returns:
+            str: The unique cluster name.
+        """
+        stable_input = f"{self.app.name}-{self.model.uuid}"
```

Comment:
is `stable_input` not already a unique name given the model uuid? Can we have multiple clusters per model?

Reply:
right, by including the `app.name`, this allows for multiple clusters in the same juju model

```sh
juju deploy k8s k8s-cluster-1
juju deploy k8s k8s-cluster-2
```